src/beacon/use-beacon-handler.hook.ts

```diff
@@ -9,13 +9,17 @@ import { useNavigation } from '../navigator/hooks/use-navigation.hook';
 import { isDefined } from '../utils/is-defined';
 import { BeaconHandler, isBeaconMessage } from './beacon-handler';
 
-export const beaconDeepLinkHandler = async (url: string | null) => {
+export const beaconDeepLinkHandler = async (
+  navigate: (modal: ModalsEnum, params: unknown) => void,
+  url: string | null
+) => {
```

Comment:
it's better to receive `url` and `onSomethingCallback` (`onBeaconConnectionStart` for example) then the `navigate` function.